apps/e2e-tests/src/process-compose/e2e.spec.ts

```diff
@@ -249,9 +241,14 @@ describe.sequential('E2E Tests with process-compose', () => {
         }
         expect(value).not.toEqual(initialFeedsInfo[id].value);
 
+        const actualRound =
+          (MAX_HISTORY_ELEMENTS_PER_FEED +
+            (round - initialFeedsInfo[id].round) -
+            1) %
+          MAX_HISTORY_ELEMENTS_PER_FEED;
         const actualData = historyData.aggregate_history[id].find(
-          feed => feed.update_number === round - initialFeedsInfo[id].round - 1,
-        );
+          feed => feed.update_number === actualRound,
+        ); // here the round counter is >= 2 (from previous test) so it is safe to subtract 1 from the update number
```

Comment:
```suggestion
        const actualData = historyData.aggregate_history[id].find(
          // In history data, the indexing of updates starts from 0.
          // Hence, we need to subtract 1 from the number of updates
          feed => feed.update_number === updatesToNetworks[network][id] - 1,
        );
```

Reply:
At this point, we have the following:
- Some updates have happened.
- At `currentFeedsInfo`, we have data for rounds = initial round + num of updates according to the sequencer.
- We need to compare updated on-chain data with what the sequencer thinks has happened ( values from `historyData` ). So we can index `historyData` with the number of updates according to the sequencer - 1 ( as indexing in history data starts from 0 )

We do not need the complicated math that calculates `actualRound` ( which in fact is the num of updates, not a round value )

Reply:
We can add one more check - one that confirms that the sequencer does not overwrite the latest round.
```ts
        const expectedNumberOfUpdates =
          (MAX_HISTORY_ELEMENTS_PER_FEED +
            (round - initialFeedsInfo[id].round)) %
          MAX_HISTORY_ELEMENTS_PER_FEED;

        expect(expectedNumberOfUpdates).toEqual(updatesToNetworks[network][id]);
```
In `expectedNumberOfUpdates`, we calculate how many updates have happened after collecting the initial feed info. When we compare it with the number of updates according to the sequencer we can confirm the sequencer has proceeded from the last update and has not rewritten it (known previous bug )